lib/src/main/java/com/auth0/android/lock/internal/configuration/Connection.java

```diff
@@ -0,0 +1,186 @@
+        }
+        if ("fair".equals(value)) {
+            return PasswordStrength.FAIR;
+        }
+        if ("low".equals(value)) {
+            return PasswordStrength.LOW;
+        }
+        return PasswordStrength.NONE;
+    }
+
+    @Override
+    public boolean requiresUsername() {
+        return booleanForKey("requires_username");
+    }
+
+    @Override
+    public boolean showSignUp() {
+        return booleanForKey("showSignup");
+    }
+
+    @Override
+    public boolean showForgot() {
+        return booleanForKey("showForgot");
+    }
+
+    @Override
+    public int getMinUsernameLength() {
+        return minUsernameLength;
+    }
+
+    @Override
+    public int getMaxUsernameLength() {
+        return maxUsernameLength;
+    }
+
+    @Override
+    public boolean isActiveFlowEnabled() {
+        return "ad".equals(getStrategy()) || "adfs".equals(getStrategy()) || "waad".equals(getStrategy());
+    }
+
+    @Override
+    public Set<String> getDomainSet() {
+        Set<String> domains = new HashSet<>();
+        String domain = getValueForKey("domain");
+        if (domain != null) {
+            domains.add(domain.toLowerCase());
+            List<String> aliases = getValueForKey("domain_aliases");
+            if (aliases != null) {
+                for (String alias : aliases) {
+                    domains.add(alias.toLowerCase());
+                }
+            }
+        }
+        return domains;
+    }
+
+    /**
+     * Creates a new Connection given a Strategy name and the map of values.
+     *
+     * @param strategy strategy name for this connection
+     * @param values   additional values associated to this connection
+     * @return a new instance of Connection. Can be either DatabaseConnection, PasswordlessConnection or OAuthConnection.
+     */
+    public static Connection connectionFor(@NonNull String strategy, Map<String, Object> values) {
+        return new Connection(strategy, values);
+    }
+
+    private void parseUsernameLength() {
+        Map<String, Object> validations = getValueForKey("validation");
+        if (validations == null || !validations.containsKey("username")) {
+            minUsernameLength = MIN_USERNAME_LENGTH;
+            maxUsernameLength = MAX_USERNAME_LENGTH;
+            return;
+        }
+        final Map<String, Object> usernameValidation = (Map<String, Object>) validations.get("username");
+        minUsernameLength = intValue(usernameValidation.get("min"));
+        maxUsernameLength = intValue(usernameValidation.get("max"));
+        if (minUsernameLength < MIN_USERNAME_LENGTH || minUsernameLength > maxUsernameLength) {
+            minUsernameLength = MIN_USERNAME_LENGTH;
+            maxUsernameLength = MAX_USERNAME_LENGTH;
+        }
+    }
+
+    private int intValue(@Nullable Object object) {
+        int value = 0;
+        try {
+            value = object == null ? 0 : (int) Double.parseDouble(String.valueOf(object));
```

Comment:
We can't fix the structure of that json dictionary as it may change in the future, so we need to set the target to a generic map. When deserializing a `Map<String, Object>` *Gson* will try to convert "string numbers" to double so not to lose any precision. That's why we need to get the double first and then remove that precision. More info [here](http://stackoverflow.com/questions/15507997/how-to-prevent-gson-from-expressing-integers-as-floats).